### ISSUE TITLE
Folder structure

### ISSUE THREAD

**Issue by distantnative** [plaidscreen/kart#1]
To get the thoughts going...

Kart core will be place in ```site/plugins/kart/```. That could include the following subfolders:

```
kart
 | - core
 | - config
 | - blueprints
 | - snippets
 | - templates
 | - languages
 | - plugins/addons
      | -  customers
      | -  emails
      | -  coupons
      | -  related-products
      | -  statistics
| - gateways
      | -  stripe

**Comment by plaidscreen** [comment]
This is great. I've been trying to think of ways to expand on it, I can't think of any, so I'm just going to run with it. Although I do feel like "gateways" should just be part of standard plugins/addons, not separate. 

Panel will be a sub-repo, like it is in Kirby.

Do you think front-end CSS/JS should go in the templates folder? Maybe a separate assets folder?

**Comment by benseitz** [comment]
separate assets folder sounds best for me

**Comment by plaidscreen** [comment]
@benseitz Certain CSS/JS seems like it should be core functionality. Maybe core/assets for the stuff developers can depend on?

I just realized it doesn't make any sense to have CSS/JS go in templates. We're mimicking Kirby's folder structure, not Wordpress's :)

**Comment by benseitz** [comment]
Just to be clear: CSS/JS for the panel plugins or the example templates?
(Maybe a starterkit with templates, blueprints and assets - but also a plainkit, so developers can start from scratch?)

**Comment by plaidscreen** [comment]
@benseitz Panel assets would be separate. But there's certain Javascript functionality I'd like to always be available to developers who are building Kart-friendly themes.

**Comment by benseitz** [comment]
The more I think about it: templates, snippets and assets shouldn't be in the site/plugins/kart folder, since they may overlap with the other templates, snippets and assets.

Not always theres only a shop on a subdomain. Think of a blog with sidebar that shows a product (okay, bad example, but you get the point). Would that be a standard kirby template or a kart template?

**Comment by benseitz** [comment]
Well, product blueprints be available in the standard kirby content structure or am I wrong?

**Comment by plaidscreen** [comment]
@benseitz Yeah - I thought originally it would be best to keep any theming elements completely out of the plugin, that includes snippets, templates, and assets. But there's going to need to be some sort of "default", and we were trying to keep it all self-contained, so I'm not really sure what's best.

**Comment by benseitz** [comment]
Copying a few files into the assets, templates, snippets and blueprint folder isn't that bad..
Maybe in cooperation with Bastian we/he could offer a starterkit download containing both kirby and kart with all the templates and assets where they belong?

**Comment by distantnative** [comment]
How about such a starterkit with basic blueprints, templates, snippets, basic css/js, that first just lies in the kart-folder, but with an install scripts that would copy it in the right places in the normal Kirby structure? And if developers wanna build it from scratch, they just don't klick the "Install Starterkit" button

**Comment by distantnative** [comment]
Also: there could be country-specific starterkits eventually, e.g. paying regard to German laws for legal texts that have to be shown on a shop website, those could also be installed with a starterkit_DE

**Comment by benseitz** [comment]
:+1: perfect solution

**Comment by plaidscreen** [comment]
So, to make this is extendable as possible, maybe we have a themes folder with a subfolder "default" that comes packaged with Kart? Then there's an area in the panel to easily install/uninstall theme files (each theme would include blueprints, templates, assets, and snippets).

Then third party developers could write Kart compatible themes that could be dropped in easily, and the Kart panel could manage the installation (if the user so desired)

**Comment by benseitz** [comment]
A themes folder sound good! Since designing a shop is quite complicated I guess this would be much more needed than for standard websites.

**Comment by distantnative** [comment]
Better for working on it: https://github.com/kirby-kart/kart/wiki/Folder-structure

**Comment by benseitz** [comment]
@distantnative isn't panel the same as plugins/order-management?

**Comment by distantnative** [comment]
@benseitz not essentially, I think the core / panel will have a very basic order management. With the plugin I thought of things like adding notes, order process history and things to the order management. Things that make it easier for a team to process orders, but nothing that needs to be in the core of Kart.

**Comment by benseitz** [comment]
But since kart should work without any panel - the panel thing should be a addon as well :)

**Comment by distantnative** [comment]
I'm not sure what you mean: ```panel```should also be moved into the ```addons``` directory?
I'd leave it similar to Kirby at the same level as the core.

**Comment by benseitz** [comment]
but the message is: everything in the addon folder can be deleted - everything in the main folder needs to stay

**Comment by benseitz** [comment]
Or we could seperate addons from panel addons? But a basic panel addon in the main folder and a more detailed order-management addon in the addon folder doesn't make sense to me.

**Comment by distantnative** [comment]
The problem with that: Even if we have the core panel, as soon as tax, shipping, stock management or coupons are added, they also need a representation in the Kart panel

**Comment by benseitz** [comment]
So we have dependencies... But on the other hand: Who uses a shop without a backend? Why not make the panel a core feature?

**Comment by plaidscreen** [comment]
Re: panel addon
I think the reason the Kirby Panel is in the root folder is so it's easily accessible from the panel/ url. Can't you rename it to whatever you want? Anyway, since ideally the Kart panel is just going to extend the kirby panel, I kind of agree with @benseitz, I think it should be in the addons folder. It can have its own triggers and filters too, so other plugins (tax, shipping, stock, etc) could just hook into those to add Panel features.

Re: themes
The more I think about it, the more I think it doesn't make sense to have any theme or template files inside of the Kart plugin. How often do you ever actually switch themes? A seperate Kart starterkit repo, which includes some sample blueprints, snippets, templates, in the standard Kirby locations, just feels the most "Kirby-like" to me. Let's keep it as barebones as possible!

**Comment by benseitz** [comment]
i agree with you @plaidscreen 
Installing 3rd party themes might get more complicated, but everyone who uses kirby knows how to paste a few files in some folders.

**Comment by distantnative** [comment]
Re: themes
True. If we could still find a really simple way for people to move all these starterkit things into the respective Kirby folders, that'd be great. But let's for now leave that aside.

**Comment by benseitz** [comment]
How about a Kart customizer like: http://getbootstrap.com/customize/
so people could select which addons and languages they need?

**Comment by texnixe** [comment]
I also think it would be best to have the templates, blueprints etc. in the normal Kirby folders, tbh, anyone who wants to set up a shop should be able to move a few files around. A customizer sounds like a great idea, shops can be anything from selling files etc. electronically to a full fledged shop with hundreds of "real" products and variations, that need to be shipped, so the requirements would be very different.
